qiskit_experiments/base_experiment.py

```diff
@@ -99,62 +70,92 @@ def __init__(
         self,
-        backend: "Backend",
+        backend: Backend,
         analysis: bool = True,
         experiment_data: Optional[ExperimentData] = None,
-        **kwargs,
+        **run_options,
     ) -> ExperimentData:
         """Run an experiment and perform analysis.
 
         Args:
             backend: The backend to run the experiment on.
-            analysis: If True run analysis on experiment data.
-            experiment_data: Optional, add results to existing experiment data.
-                             If None a new ExperimentData object will be returned.
-            kwargs: keyword arguments for self.circuit, qiskit.transpile, and backend.run.
+            analysis: If True run analysis on the experiment data.
+            experiment_data: Optional, add results to existing
+                experiment data. If None a new ExperimentData object will be
+                returned.
+            run_options: backend runtime options used for circuit execution.
 
         Returns:
-            ExperimentData: the experiment data object.
+            The experiment data object.
         """
-        # NOTE: This method is intended to be overriden by subclasses if required.
-
         # Create new experiment data
         if experiment_data is None:
             experiment_data = self.__experiment_data__(self, backend=backend)
 
-        # Filter kwargs
-        run_options = self.__run_defaults__.copy()
-        circuit_options = {}
-        for key, value in kwargs.items():
-            if key in _TRANSPILE_OPTIONS or key in self._circuit_options:
-                circuit_options[key] = value
-            else:
-                run_options[key] = value
-
-        # Generate and run circuits
-        circuits = self.transpiled_circuits(backend, **circuit_options)
+        # Generate and transpile circuits
+        circuits = transpile(self.circuits(backend), backend, **self.transpile_options.__dict__)
+
+        # Run circuits on backend
+        run_opts = copy.copy(self.run_options)
+        run_opts.update_options(**run_options)
+        run_opts = run_opts.__dict__
+
         if isinstance(backend, LegacyBackend):
-            qobj = assemble(circuits, backend=backend, **run_options)
+            qobj = assemble(circuits, backend=backend, **run_opts)
             job = backend.run(qobj)
         else:
-            job = backend.run(circuits, **run_options)
+            job = backend.run(circuits, **run_opts)
 
         # Add Job to ExperimentData
         experiment_data.add_data(job)
 
         # Queue analysis of data for when job is finished
         if analysis and self.__analysis_class__ is not None:
-            # pylint: disable = not-callable
-            self.__analysis_class__().run(experiment_data, **kwargs)
+            self.run_analysis(experiment_data)
 
         # Return the ExperimentData future
         return experiment_data
 
+    def run_analysis(self, experiment_data, **options) -> ExperimentData:
```

Comment:
A few lines above is written:
```python
             self.run_analysis(experiment_data)
```
i.e. I see no options. Is this not inconsistent with the method signature? Do we need `**options` here?

Reply:
The direct call to `run_analysis` allows overriding the experiments stored analysis options with the kwarg ones (like run does with runtime options). At the moment for split execution/analysis the following are basically equivalent:

```python
# 1
exp.set_analysis_options(**analysis_opts)
expdata = exp.run(backend, analysis=False)
exp.run_analysis(expdata)

# 2
expdata = exp.run(backend, analysis=False)
exp.run_analysis(expdata, **analysis_opts)
```

Removing the kwargs from run_analysis would remove option 2. Opt 2 is not strictly necessary, but I thought it might be convenient in cases where you are running analysis separately to the main execution, or analyze expdata loaded from a DB.

Reply:
That makes sense. Thanks. At the risk of being pedantic, can we document this in the docstring to make the two usage options clear and that an analysis can be run outside the context of an experiment?